stablehlo/dialect/VhloOps.td

```diff
@@ -61,672 +53,807 @@ class VHLO_Op<string mnemonic, string minVersion, string maxVersion, list<Trait>
-    VHLO_AnyAttr:$infeed_config,
-    OptionalAttr<VHLO_AnyAttr>:$layout
+    Variadic<VHLO_AnyType>:$inputs,
+    VHLO_AnyAttr:$call_target_name,
+    OptionalAttr<VHLO_AnyAttr>:$has_side_effect,
+    VHLO_AnyAttr:$backend_config,
+    OptionalAttr<VHLO_AnyAttr>:$api_version,
+    OptionalAttr<VHLO_AnyAttr>:$called_computations,
+    OptionalAttr<VHLO_AnyAttr>:$operand_layouts,
+    OptionalAttr<VHLO_AnyAttr>:$result_layouts
   );
-  let results = (outs Variadic<VHLO_AnyType>);
+  let results = (outs Variadic<VHLO_AnyType>:$results);
 }
 
-def VHLO_OutfeedOpV1 : VHLO_Op<"outfeed", "0.3.0", "current"> {
+// TODO(#1187): api_version is different between VHLO and the StableHLO spec:
+// in VHLO it's an enum, in the spec it's an si32.
+// TODO(#629): operand_layouts/result_layouts are not part of the spec.
+// TODO(#740): output_operand_aliases is not part of the spec.
+// CustomCallOp has proven to be one of the trickiest ops to fully spec.
+// We're aiming to address all these todos by the release of StableHLO v1.0.
+def VHLO_CustomCallOpV2: VHLO_Op<"custom_call_v2", "0.4.0", "current"> {
   let arguments = (ins
     Variadic<VHLO_AnyType>:$inputs,
-    VHLO_AnyType:$token,
-    VHLO_AnyAttr:$outfeed_config
+    VHLO_AnyAttr:$call_target_name,
+    OptionalAttr<VHLO_AnyAttr>:$has_side_effect,
+    VHLO_AnyAttr:$backend_config,
+    OptionalAttr<VHLO_AnyAttr>:$api_version,
+    OptionalAttr<VHLO_AnyAttr>:$called_computations,
+    OptionalAttr<VHLO_AnyAttr>:$operand_layouts,
+    OptionalAttr<VHLO_AnyAttr>:$result_layouts,
+    OptionalAttr<VHLO_AnyAttr>:$output_operand_aliases
   );
-  let results = (outs VHLO_AnyType);
+  let results = (outs Variadic<VHLO_AnyType>:$results);
 }
 
-def VHLO_SendOpV1 : VHLO_Op<"send", "0.3.0", "current"> {
-  let arguments = (ins
-    Variadic<VHLO_AnyType>:$inputs,
-    VHLO_AnyType:$token,
-    VHLO_AnyAttr:$channel_handle,
-    OptionalAttr<VHLO_AnyAttr>:$is_host_transfer
-  );
-  let results = (outs VHLO_AnyType);
+def VHLO_DivOpV1 : VHLO_Op<"divide", "0.3.0", "current"> {
+  let arguments = (ins VHLO_AnyType:$lhs, VHLO_AnyType:$rhs);
+  let results = (outs VHLO_AnyType:$result);
 }
 
-def VHLO_RecvOpV1 : VHLO_Op<"recv", "0.3.0", "current"> {
+def VHLO_DotGeneralOpV1 : VHLO_Op<"dot_general", "0.3.0", "current"> {
   let arguments = (ins
-    VHLO_AnyType:$token,
-    VHLO_AnyAttr:$channel_handle,
-    OptionalAttr<VHLO_AnyAttr>:$is_host_transfer
+    VHLO_AnyType:$lhs,
+    VHLO_AnyType:$rhs,
+    VHLO_AnyAttr:$dot_dimension_numbers,
+    OptionalAttr<VHLO_AnyAttr>:$precision_config
   );
-  let results = (outs Variadic<VHLO_AnyType>);
-}
-
-// Parallelism related op definitions.
-def VHLO_ReplicaIdOpV1 : VHLO_Op<"replica_id", "0.3.0", "current"> {
-  let results = (outs VHLO_AnyType);
+  let results = (outs VHLO_AnyType:$result);
 }
 
-def VHLO_PartitionIdOpV1 : VHLO_Op<"partition_id", "0.4.0", "current"> {
-  let results = (outs VHLO_AnyType);
+// TODO(#3): DotOp is not part of the StableHLO spec.
```

Comment:
Should we deprecate these in the near future?

Deprecation in this case refers to failing on the `StableHLO-->VHLO` conversions for these ops, while still supporting the `VHLO_old->VHLO_curr->StableHLO` pathways.

Reply:
Yeah, this is what I wanted to discuss this later today - what tickets we would want to frontload to ensure a smooth StableHLO v1.0 release down the line. Let's chat offline.